x-pack/plugins/data_visualizer/public/application/index_data_visualizer/hooks/esql/use_esql_field_stats_data.ts

```diff
@@ -0,0 +1,170 @@
+import { useCancellableSearch } from '@kbn/ml-cancellable-search';
+import type { DataStatsFetchProgress, FieldStats } from '../../../../../common/types/field_stats';
+import { useDataVisualizerKibana } from '../../../kibana_context';
+import { getInitialProgress, getReducer } from '../../progress_utils';
+import { isESQLQuery, getSafeESQLLimitSize } from '../../search_strategy/requests/esql_utils';
+import type { Column } from './use_esql_overall_stats_data';
+import { getESQLNumericFieldStats } from '../../search_strategy/esql_requests/get_numeric_field_stats';
+import { getESQLKeywordFieldStats } from '../../search_strategy/esql_requests/get_keyword_fields';
+import { getESQLDateFieldStats } from '../../search_strategy/esql_requests/get_date_field_stats';
+import { getESQLBooleanFieldStats } from '../../search_strategy/esql_requests/get_boolean_field_stats';
+import { getESQLTextFieldStats } from '../../search_strategy/esql_requests/get_text_field_stats';
+
+export const useESQLFieldStatsData = <T extends Column>({
+  searchQuery,
+  columns: allColumns,
+  filter,
+  limitSize,
+}: {
+  searchQuery?: AggregateQuery;
+  columns?: T[];
+  filter?: QueryDslQueryContainer;
+  limitSize?: string;
+}) => {
+  const [fieldStats, setFieldStats] = useState<Map<string, FieldStats>>();
+
+  const [fetchState, setFetchState] = useReducer(
+    getReducer<DataStatsFetchProgress>(),
+    getInitialProgress()
+  );
+
+  const {
+    services: {
+      data,
+      notifications: { toasts },
+    },
+  } = useDataVisualizerKibana();
+
+  const { runRequest, cancelRequest } = useCancellableSearch(data);
+
+  useEffect(
+    function updateFieldStats() {
+      let unmounted = false;
+
+      const fetchFieldStats = async () => {
+        cancelRequest();
+
+        if (!isESQLQuery(searchQuery) || !allColumns) return;
+
+        setFetchState({
+          ...getInitialProgress(),
+          isRunning: true,
+          error: undefined,
+        });
+        try {
+          // By default, limit the source data to 100,000 rows
+          const esqlBaseQuery = searchQuery.esql + getSafeESQLLimitSize(limitSize);
+
+          const totalFieldsCnt = allColumns.length;
+          const processedFieldStats = new Map<string, FieldStats>();
+
+          function addToProcessedFieldStats(stats: Array<FieldStats | undefined>) {
+            if (!unmounted) {
+              stats.forEach((field) => {
+                if (field) {
+                  processedFieldStats.set(field.fieldName!, field);
+                }
+              });
+              setFetchState({
+                loaded: (processedFieldStats.size / totalFieldsCnt) * 100,
+              });
+            }
+          }
+          setFieldStats(processedFieldStats);
+
+          const aggregatableFieldsChunks = chunk(allColumns, 25);
+
+          for (const columns of aggregatableFieldsChunks) {
+            // GETTING STATS FOR NUMERIC FIELDS
+            await getESQLNumericFieldStats({
```

Comment:
Shall me combine all of these request (`getESQLNumericFieldStats`, `getESQLBooleanFieldStats`, etc.) with `Promise.all`?